Add an Endpoint to List DataONE Package Contents

ThomasThelen opened the pull request:
This PR corresponds to issue #53 . A new endpoint, called `listFiles` was created which takes the same argument as `lookup`. I split some of the code that was already present in the file into smaller functions to avoid re-writing it in my changes. Also, ignore the `bla` commit-the changes were undone in commit 3616ce3.

Note about the tests: It looks like codecov isn't taking the functions in `dataone_register.py` into consideration, which is why the coverage is so low. This may be because we are mocking with `httmock`

Below is an example of the extent of coverage (red highlighting is code that wasn't "hit").

https://codecov.io/gh/ThomasThelen/girder_wholetale/src/413df2a2c7dea9b4de84f5d1f3fc3b342008eaba/server/dataone_register.py

amoeba commented:
Cool, I'll take a look at this in a bit today.

amoeba commented on requirements.txt:
```diff
@@ -2,3 +2,4 @@ rdflib
 cryptography
 tornado
 celery[redis]
+
```
Could you nuke this commit and force push your fork so this PR stays cleaner?

amoeba commented on requirements.txt:
```diff
@@ -1,5 +1,4 @@
 rdflib
 cryptography
 tornado
-celery[redis]
```
also please nuke this commit since these changes don't relate to the pr

amoeba commented on plugin_tests/dataone_test.py:
```diff
@@ -10,6 +10,7 @@
 from girder.constants import ROOT_DIR
 from girder.api.rest import RestException
 
+base.TestCase.maxDiff = None
```
is this needed? https://docs.python.org/3/library/unittest.html#unittest.TestCase.maxDiff

amoeba commented on server/rest/repository.py:
```diff
@@ -49,7 +49,30 @@
+    'required': ['name', 'size'],
+    'example': {
+        'name': 'urn:uuid:42969280-e11c-41a9-92dc-33964bf785c8',
+        'size': '10.5063/F1Z899CZ'
```
this is probably supposed to be an integer instead of a DOI string

amoeba commented on server/dataone_register.py:
```diff
@@ -242,3 +244,41 @@ def D1_lookup(path):
+    files = list()
+    for doc in docs:
+        data_map = {
+            'name': doc.get('title', doc.get('identifier')),
```
I think this should be `doc.get('fileName', doc.get('identifier')`. `title` is only for metadata Objects anyway.

amoeba commented:
This looks pretty good. I made some in-line comments. It looks like this implements just enough to get the list of files to populate the Dashboard, but doesn't actually do the per-file registration, right? And: Does this handle hierarchical packages yet?

@Xarthisius, from an architectural/API perspective, are you cool with adding another API method for this? If I'm right about my question above, it looks like we'll to modify the behavior of the `/dataset/register` method so it can register either an entire package (current ability) or just the files from a package the user has selected here. Does that sound about right?

Xarthisius commented:
@amoeba sounds good to me! We're going to need such endpoint anyway for "importing Tales". If we keep that in mind, maybe it would be worthwhile to add 'location' to `fileMap` along with the other things that are necessary from Tale serialization POV.

ThomasThelen commented:
1. Reverted the two extra commits.
2. Removed the 
3. Added a field for the file's parent package and ID to the `dataMap` object that is sent to the UI

ThomasThelen commented:
This PR is pretty messy so I'm closing it without a merge. Refer to PR #57 for a clean version
